aggregator/apps/api/v1/stitcher.py

```diff
@@ -56,14 +57,66 @@ def __init__(self, ballots):
         self.cancelled_ballots = self.get_cancelled_ballots()
 
     def get_cancelled_ballots(self):
-        return [b for b in self.ballots if b["cancelled"]]
+        return [b for b in self.ballots if b.get("cancelled", False)]
 
     def get_metadata_by_key(self, key):
         for b in self.ballots:
-            if b["metadata"] and key in b["metadata"]:
+            if b.get("metadata") and key in b["metadata"]:
                 return b["metadata"][key]
         return None
 
+    def get_ballot_cancellation_reason(self, ballot: Dict) -> Optional[str]:
+        """
+        Given a cancelled ballot, determine the reason for cancellation
+        and return it as a friendly string :) (or None)
+        :param ballot: Dict representing a ballot object
+        :return: str or None
+        """
+        ballot_candidate_count = len(ballot.get("candidates", []))
+        ballot_seats_contested = ballot.get("seats_contested", 0)
+
+        if ballot_candidate_count == ballot_seats_contested:
+            return "Uncontested election with equal candidates to seats"
+        elif ballot_candidate_count < ballot_seats_contested:
+            if ballot_candidate_count != 0:
+                return (
+                    "Uncontested election with fewer candidates than seats but not zero"
```

Comment:
Can this be "Uncontested election with fewer candidates than seats" and below be "Uncontested election with no candidates"?

(Sorry I know this is widget stuff, but I'm just re-reading the logic and I think this is clearer)